### PR TITLE
ci: parallelize uploads in `publish-docs`

### PR DESCRIPTION
**Pull request by coryan**
Use the same ideas as `check-api`. Move the retry loop outside the function so we can see the output when a retry is needed.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12189)

### CODE REVIEW

**Comment by coryan** [comment]
Does not make a huge difference, but a few seconds saved are nothing to sneeze at.

Most of the time is spent in Doxygen, parsing the code.  All CPUs are busy, there is no more parallelism to squeeze here, the next improvement will come from sharding.

**Review by devbww — APPROVED** [review]

File: ci/cloudbuild/builds/publish-docs.sh [hunk 1]
```diff
@@ -103,40 +103,49 @@ export LANG=C.UTF-8
...
 
+echo common "${FEATURE_LIST[@]}" | xargs -P "$(nproc)" -n 1 \
+  bash -c "TIMEFORMAT=\"\${0} completed in %0lR\";
+           time ci/retry-command.sh 3 120 stage_docfx \"\${0}\" \"${docfx_bucket}\" cmake-out \"cmake-out/\${0}.docfx.log\""
+
+errors=0
 for feature in common "${FEATURE_LIST[@]}"; do
-  if [[ "${feature}" == "experimental-storage-grpc" ]]; then continue; fi
-  if [[ "${feature}" == "grafeas" ]]; then continue; fi
-  if [[ "${feature}" == "experimental-opentelemetry" ]]; then feature="opentelemetry"; fi
```
Comment (dbolduc) [inline ci/cloudbuild/builds/publish-docs.sh:131]: Don't we still want these?

**Review by dbolduc — CHANGES_REQUESTED** [review]

File: ci/cloudbuild/builds/publish-docs.sh (hunk 1 above)
Comment (dbolduc) [reply ci/cloudbuild/builds/publish-docs.sh:131]: I think every time we loop over `FEATURE_LIST` we want to
- skip `experimental-storage-grpc`
- skip `grafeas`
- transform `experimental-opentelemetry` -> `opentelemetry`

Comment (coryan) [reply ci/cloudbuild/builds/publish-docs.sh:131]: I fixed `experimental-opentelemetry` -> `opentelemetry`. The other changes are not needed for the reasons explained in the new comments.

**Pull request merged**